server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -431,15 +572,13 @@ private void assertConsistentStates() {
 
         private void updateCommittedStates() {
             for (final ClusterNode clusterNode : clusterNodes) {
-                Optional<ClusterState> committedState = clusterNode.coordinator.getLastCommittedState();
-                if (committedState.isPresent()) {
-                    ClusterState storedState = committedStatesByVersion.get(committedState.get().getVersion());
-                    if (storedState == null) {
-                        committedStatesByVersion.put(committedState.get().getVersion(), committedState.get());
-                    } else {
-                        assertEquals("expected " + committedState.get() + " but got " + storedState,
-                            value(committedState.get()), value(storedState));
-                    }
+                ClusterState applierState = clusterNode.coordinator.getApplierState();
+                ClusterState storedState = committedStatesByVersion.get(applierState.getVersion());
```

Comment:
I guess we should call this method `updateAppliedStates()` and the field should be `appliedStatesByVersion`.

Reply:
`appliedStates` is not fully correct (those states might not be applied yet, they are just sent to the applier (which is not what we actually want to check here)). `applierStates` is a little weird, so I would like to stick to committedStates. Every state that is sent to the applier is a committed state, and what we're interested in here is checking that committed states are uniquely defined by their term and version.